### ISSUE TITLE
basic_string::resize error

### ISSUE THREAD

**Issue by ajing** [bwilbertz/RLightGBM#8]
I got the following error. It might come from lgbm C function, but do you have any hint about what is happening?

![image](https://cloud.githubusercontent.com/assets/1566338/21740611/a885e9d0-d473-11e6-95fc-c0900717b211.png)

**Comment by bwilbertz** [comment]
Hard to say. Does it work if you use only the first two arguments (leaving the other as default)?

**Comment by ajing** [comment]
fixed because of the data set..

**Comment by ajing** [comment]
OK, now I can reproduce the error. It happens when I save the model object as RDS and then load the RDS file.

Here is the example:

`
library(caret)
library(RLightGBM)
library(xgboost)

data(agaricus.train)

model <-caretModel.LGBM()

lgbm_trcontrol_1 = trainControl(
  method = "cv",
  number = 3,
  verboseIter = TRUE,
  returnData = TRUE,
  returnResamp = "all",                                                        # save losses across all models
  classProbs = TRUE,                                                           # set to TRUE for AUC to be computed
  summaryFunction = twoClassSummary,
  allowParallel = TRUE,
  savePredictions = TRUE,
  selectionFunction = "oneSE"
)

lgbm_train_1 = train(
  x = data.matrix(agaricus.train$data[, colnames(agaricus.train$data) %in% colnames(agaricus.train$data)[1:20]]),
  y = factor(ifelse(agaricus.train$label == 0, "B", "M")),
  trControl = lgbm_trcontrol_1,
  tuneLength = 2,
  search = "random",
  method = model,
  metric = "ROC"
)

y.pred <- predict(lgbm_train_1, data.matrix(agaricus.train$data[, colnames(agaricus.train$data) %in% colnames(agaricus.train$data)[1:20]]), type = "prob")

saveRDS(data.matrix(agaricus.train$data[, colnames(agaricus.train$data) %in% colnames(agaricus.train$data)[1:20]]), file = "tmpdata.RDS")
saveRDS(lgbm_train_1, file = "tmp.RDS")`

Then, open another R terminal, do

`dta <- readRDS("tmpdata.RDS");
mod <- readRDS("tmp.RDS");
predict(mod, dta)`

Loading required package: RLightGBM
Loading required package: plyr
Error in lgbm.booster.predict(modelFit$booster.handle, newdata, 0L, modelFit$num_models) :
  basic_string::resize

**Comment by bwilbertz** [comment]
Hi, 

you cannot serialize the LightGBM model using saveRDS (or any other R mechanism) because the `handle.booster` is just a C pointer.

Use `lgbm.booster.save` and `lgbm.booster.load` to save and restore what you have under `lgbm_train_1$finalModel$handle.booster`.
